src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -1867,7 +1871,10 @@
     <Field Name="Symbol" Type="LambdaSymbol" Null="disallow"/>
     <Field Name="Type" Type="TypeSymbol?" Override="true"/>
     <Field Name="Body" Type="BoundBlock"/>
-    <Field Name="Diagnostics" Type="ImmutableArray&lt;Microsoft.CodeAnalysis.Diagnostic&gt;" />
+    <!-- PROTOTYPE(UsedAssemblyReferences): Remove Null="allow" once code generator is adjusted to 
+                                            handle ImmutableBindingDiagnostic specially.
+    -->
+    <Field Name="Diagnostics" Type="ImmutableBindingDiagnostic&lt;AssemblySymbol&gt;" Null="allow"/>
```

Comment:
>allow [](start = 92, length = 5)

Did the `ValueType` entry not work?  Can it be made to work?

Reply:
> Did the ValueType entry not work?  Can it be made to work?

No, this structure contains ImmutableArrays and those are also handled specially by the code generator. Value types in general do not have a concept of "Null", however ImmutableArrays have a concept of IsDefault, which is used for the purpose of "Null" checks by the generator.  

---
In reply to: [361750921](https://github.com/dotnet/roslyn/pull/40560#discussion_r361750921) [](ancestors = 361750921)